src/main/webapp/cluster-view/cluster-view.controller.js

```diff
@@ -1,26 +1,170 @@
 'use strict';
 
 angular.module('managementConsole')
-    .controller('ClusterViewCtrl', [
+  .controller('ClusterViewCtrl', [
     '$scope',
     '$stateParams',
     '$state',
+    '$q',
     'modelController',
-    function ($scope, $stateParams, $state, modelController) {
-            if (!modelController.isAuthenticated()) {
-                $state.go('/logout');
+    function ($scope, $stateParams, $state, $q, modelController) {
+      if (!modelController.isAuthenticated()) {
+        $state.go('/logout');
+      }
+      $scope.shared = {
+        currentCollection: 'caches'
+      };
+      $scope.clusters = modelController.getServer().getClusters();
+      $scope.currentCluster = modelController.getServer().getCluster($scope.clusters, $stateParams.clusterName);
+
+      $scope.$watch('currentCluster', function (currentCluster) {
+        if (currentCluster && currentCluster.name !== $stateParams.clusterName) {
+          $state.go('clusterView', {
+            'clusterName': currentCluster.name
+          });
+        }
+      });
+
+      $scope.isNodesView = function (){
+        return $scope.shared.currentCollection === 'nodes';
+      };
+
+      $scope.isCachesView = function (){
+        return $scope.shared.currentCollection === 'caches';
+      };
+
+      $scope.zoomedIn = false;
+
+      var zoomChanged = function () {
+        var slid = $("#zoom");
```

Comment:
Ok I finally have the thing running! I took a look at interface, and if I am correct, the whole filtering goes like this: we have certain values (average read time, write time and so on) and we have sliders which modify this values. When values modify, we do adjust `show` attribute of nodes/caches.

Slider is a highly reusable component with view (html) and logic (js), and basically all it does is output a pair of integers. So instead of having all this slider logic in this controller, we should make it a directive, and we would use it something like this in `cluster-view.html`:
```
<slider ng-model="avgWriteTime" max="maxAvgWriteTime" min="minAvgWriteTime"></slider>
<slider ng-model="avgReadTime" max="maxAvgReadTime" min="minAvgReadTime"></slider>
...
```
Then, to trigger filtering based on changes of filter values, we would use `$watch` in controller to adjust the `show` attributes of nodes/caches. Awesome thing here is that we do not really care who is changing filter values: it can be sliders or anything else! So basically all goes around filter values: we have sliders changing them, and controller watching them and reacting to change by doing filtering.

After this change there should be much less code in the controller.

About slider directive: you can either find some existing angular directive (like https://github.com/seiyria/angular-bootstrap-slider that was used so far -> why are we not using it anymore?), or you can wrap ui-slider in your own directive, from what I see it should be really easy to do, basically you create a new directive, move the UiSlider code there to the right places, expose attributes and that is it -> good opportunity to learn more about directives.